scrapy_autounit/cli.py

```diff
@@ -42,36 +37,89 @@ def __init__(self, parser):
+                self._error("No recorded data found for spider '{}'".format(self.spider))
+
+            if self.callback:
+                self.callback_dir = os.path.join(self.callbacks_dir, self.callback)
+                if not os.path.isdir(self.callback_dir):
+                    self._error(
+                        "No recorded data found for callback "
+                        "'{}' from '{}' spider".format(self.callback, self.spider))
+
+                if self.fixture:
+                    self.fixture_path = os.path.join(self.callback_dir, self.parse_fixture_arg())
+                    if not os.path.isfile(self.fixture_path):
+                        self._error("Fixture '{}' not found".format(self.fixture_path))
+
+    def _error(self, msg):
+        print(msg)
+        sys.exit(1)
 
-        if not os.path.isdir(self.spider_dir):
-            self.error(
-                "No recorded data found "
-                "for spider '{}'".format(self.spider))
+    def _walk(self, root):
+        for _, subdirs, _ in os.walk(root):
+            for subdir in subdirs:
+                if subdir == '__pycache__':
+                    continue
+                yield subdir
 
+    def _get_callbacks_dir(self, spider):
         extra_path = self.settings.get('AUTOUNIT_EXTRA_PATH') or ''
-        self.callback_dir = os.path.join(
-            self.spider_dir, extra_path, self.callback)
-
-        if not os.path.isdir(self.callback_dir):
-            self.error(
-                "No recorded data found for callback "
-                "'{}' from '{}' spider".format(self.callback, self.spider))
-
-        if self.fixture:
-            self.fixture_path = os.path.join(
-                self.callback_dir, self.parse_fixture_arg())
-            if not os.path.isfile(self.fixture_path):
-                self.error("Fixture '{}' not found".format(self.fixture_path))
-
-    def error(self, msg):
-        print(msg)
-        sys.exit(1)
+        return os.path.join(self.tests_dir, spider, extra_path)
+
+    def _get_spider_fixtures(self, callbacks_dir):
+        fixtures = []
+        for callback in self._walk(callbacks_dir):
+            target = os.path.join(callbacks_dir, callback, '*.bin')
+            fixtures.extend(glob(target))
+        return fixtures
+
+    def _from_legacy_fixture(self, recorded):
+        encoding = recorded['encoding']
+        old = pickle.loads(recorded['data'], encoding=encoding)
+        new = Cassette()
+        new.spider_name = old['spider_name']
+        new.middlewares = old['middlewares']
+        new.included_settings = old['settings']
+        new.python_version = old.get('python_version', sys.version_info.major)
+        new.request = old['request']
+        new.response = old['response']
+        new.init_attrs = {}
+        new.input_attrs = old.get('spider_args_in') or old.get('spider_args') or {}
+        new.output_attrs = old.get('spider_args_out', {})
+        new.output_data = old['result']
```

Comment:
Could you pass these attributes when creating the object?

Reply:
Yup, agree. Missed this instantiation after the \_\_init\_\_ change.

Reply:
Solved in https://github.com/scrapinghub/scrapy-autounit/pull/81/commits/e93beed29b50ed5a801c7a085bc9a48599b4bcb2